scripts/install-nix-from-closure.sh

```diff
@@ -1,149 +1,618 @@
-    $nix/bin/nix-channel --update nixpkgs
-fi
-
-added=
-if [ -z "$NIX_INSTALLER_NO_MODIFY_PROFILE" ]; then
-
-    # Make the shell source nix.sh during login.
-    p=$HOME/.nix-profile/etc/profile.d/nix.sh
-
-    for i in .bash_profile .bash_login .profile; do
-        fn="$HOME/$i"
-        if [ -w "$fn" ]; then
-            if ! grep -q "$p" "$fn"; then
-                echo "modifying $fn..." >&2
-                echo "if [ -e $p ]; then . $p; fi # added by Nix installer" >> "$fn"
+
+    if [ -d "$dest" ]; then
+        print "Directory $dest already exists. Skipping creation of $dest"
+    fi
+
+}
+
+installNix() {
+
+    # In this function, if some action makes any change to the system, or does err Exit - it's need to be wrapped in a function, ex func(){...}. Inside func() you start with setup of traps, before errors/actions happen, so and according revert function be launched: 'trap funcRevertOnCondition condition', and funcRevertOnCondition is described right after trap.
+    # How all actions operate inside this function:
+    #**************************************************
+    #
+    #    installNix() {
+    #
+    #        func() {
+    #
+    #            trap 'funcRevertOnCondition $LINENO $?' condition
+    #            funcRevertOnCondition() {
+    #                errorRevert "Received Error signal from line $1"
+    #                # revert action
+    #                >&2 contactme # This function gives contact information
+    #                exit "$?"
+    #            }
+    #
+    #            # action of functions
+    #            # 'exit SING # to the trap that catches exit SIGN'
+    #        }
+    #
+    #        func
+    #
+    #    }
+    #
+    #**************************************************
+
+    # Function must have revert procedure. In case of any problem - function quits without artifacts/changes to the system
+    # Nix is pure, and claims to be transnational, so let's show people clean transnational installation
+
+    createDest() {
+
+        trap destCreateRevert 1
+
+        destCreateRevert() {
+            # all means failed, Exit!
+            error "
+
+    Can't get root access to create $dest folder. All variants failed.
+    Exiting...
+    "
+        }
+
+        # Shell parses variable with current user, so value under sudo/su not changes.
+        # It is const, but don't move it above USER set/checks.
+        readonly cmd="mkdir -m 0755 $dest && chown $USER $dest"
+
+        # Trying to create $dest by all means
+
+        # For case user is root, or has access to create $cmd. Else -> do 'then' section
+        if sh -c "$cmd" > /dev/null 2>&1 ; then
+            print "Creating $dest directory by running $cmd using $USER rights"
+        else
+            if which sudo > /dev/null 2>&1 || type sudo > /dev/null 2>&1 ; then
+                print "Creating $dest directory by running $cmd using sudo:"
+                if sudo -n true 2> /dev/null ; then
```

Comment:
This seems pointless? Just run sudo if it's available…

Reply:
If we launch "sudo" to try if it's available, it asks for password.

It is not passible to aluch sudo to check something, without sudo asking for password.

1. How you going to display: "Creating $dest directory by running $cmd using sudo:" before using "sudo" then?
2. Or if we display without check, we print "Creating $dest directory by running $cmd using sudo:" and then fail miserably, because `sudo` is not available?

Reply:
Either way, you're running sudo even if the user has no sudo permissions. You can't even check the permissions without the password. So I propose you just see if ``[ -z `type -p sudo` ]`` and then say "Trying with sudo…"

Reply:
Done.

Used `command -v`. In: https://github.com/NixOS/nix/pull/1565/commits/e2b9650833e45fba2a8b70086e49b71c6e60624b